### PR TITLE
Contrail lab2

### PR DESCRIPTION
**Pull request by k-ayache**

### CODE REVIEW

File: docker-compose/consul-data/consul-config_opencontrail_data.json [hunk 1]
```diff
@@ -1,10 +1,45 @@
...
+    "ttl_max": 60
+  },
+  "schema_transformer": {
+    "log_level": "SYS_DEBUG",
+    "sandesh_send_rate_limit": 100
+  },
+  "svc_monitor": {
+    "check_service_interval": 500,
+    "sandesh_send_rate_limit": 100,
+    "driver": "driver = svc_monitor.scheduler.vrouter_scheduler.RandomScheduler"
```
Comment (eonpatapon) [inline docker-compose/consul-data/consul-config_opencontrail_data.json]: We should simply have: 

"driver": "svc_monitor.scheduler.vrouter_scheduler.RandomScheduler"

File: docker-compose/consul-data/consul-config_opencontrail_data.json [hunk 2]
```diff
@@ -1,9 +1,45 @@
 {
-    "log_level": "SYS_INFO",
-    "api": {
-        "log_level": "SYS_DEBUG"
-    },
-    "schema_transformer": {
-        "log_level": "SYS_DEBUG"
+  "log_level": "SYS_INFO",
+  "user_ifmap": "ifmap",
```
Comment (eonpatapon) [inline docker-compose/consul-data/consul-config_opencontrail_data.json]: Rename to "ifmap_user"

File: docker-compose/consul-data/consul-config_opencontrail_data.json [hunk 3]
```diff
@@ -1,9 +1,45 @@
...
+  "api": {
+    "log_level": "SYS_DEBUG",
+    "contrail_extensions_enabled": "false",
+    "sandesh_send_rate_limit": 100,
+    "keystone_resync_workers": 10,
+    "keystone_resync_interval_secs": "86400",
+    "vnc_connection_cache_size": 128,
+    "max_request_size": "2097152",
+    "quota": {
```
Comment (eonpatapon) [inline docker-compose/consul-data/consul-config_opencontrail_data.json]: You should use the same quota names as in the config file

File: pkgs/docker-images/config/contrail/default.nix [hunk 4]
```diff
@@ -177,8 +187,25 @@ in rec {
         IFMAP_SERVER = {
           ifmap_listen_ip = containerIP;
           ifmap_listen_port = services.ifmap.port;
-          ifmap_credentials = "ifmap:" + secret "ifmap_password";
+          ifmap_credentials = ''{{$opencontrail.user_ifmap}}'' + ":" + secret "ifmap_password";
+        };
+        QUOTA = {
```
Comment (eonpatapon) [inline pkgs/docker-images/config/contrail/default.nix]: Maybe QUOTA = ''{{$opencontrail.api.quota}}''; works ?

File: docker-compose/consul-data/consul-config_opencontrail_data.json [hunk 5]
```diff
@@ -1,9 +1,45 @@
...
-        "log_level": "SYS_DEBUG"
-    },
-    "schema_transformer": {
-        "log_level": "SYS_DEBUG"
+  "log_level": "SYS_INFO",
+  "user_ifmap": "ifmap",
+  "api": {
+    "log_level": "SYS_DEBUG",
+    "contrail_extensions_enabled": "false",
+    "sandesh_send_rate_limit": 100,
```
Comment (nlewo) [inline docker-compose/consul-data/consul-config_opencontrail_data.json]: I would prefer to minimize the number of keys we have in Consul, ie. by setting these key/values at image build time. Basically, only values that are different across environments or secret should be in Consul. Some other key could be added (debug mode for instance) but requires some specific discussions.

File: pkgs/docker-images/config/contrail/default.nix [hunk 6]
```diff
@@ -177,8 +187,25 @@ in rec {
         IFMAP_SERVER = {
           ifmap_listen_ip = containerIP;
           ifmap_listen_port = services.ifmap.port;
-          ifmap_credentials = "ifmap:" + secret "ifmap_password";
+          ifmap_credentials = "api-server" + ":" + secret "ifmap_password";
+        };
+        QUOTA = {
+          virtual_network = 200;
+          subnet = 200;
+          virtual_machine_interface = "1000";
```
Comment (eonpatapon) [inline pkgs/docker-images/config/contrail/default.nix]: Use number

File: pkgs/docker-images/config/contrail/default.nix [hunk 7]
```diff
@@ -158,16 +159,25 @@ in rec {
...
           listen_port = services.api.port;
 
           disc_server_ip = services.discovery.dns;
           disc_server_port = services.discovery.port;
 
+          vnc_connection_cache_size = 128;
+
           auth = "keystone";
           aaa_mode = "cloud-admin";
+          multi_tenancy             = "True";
```
[the remark this answers is not in the document]
Comment (eonpatapon) [reply pkgs/docker-images/config/contrail/default.nix]: Yes

File: pkgs/docker-images/config/contrail/default.nix [hunk 8]
```diff
@@ -158,16 +159,25 @@ in rec {
       headers = catalogOpenstackHeader;
       conf = {
         DEFAULTS = {
+          keystone_resync_workers = 10;
+          keystone_resync_interval_secs = "86400";
```
Comment (eonpatapon) [inline pkgs/docker-images/config/contrail/default.nix]: Use number

File: pkgs/docker-images/config/contrail/default.nix [hunk 9]
```diff
@@ -158,16 +159,25 @@ in rec {
...
           disc_server_port = services.discovery.port;
 
+          vnc_connection_cache_size = 128;
+
           auth = "keystone";
           aaa_mode = "cloud-admin";
+          multi_tenancy             = "True";
+          list_optimization_enabled = "True";
+          apply_subnet_host_routes  = "True";
+          max_request_size = "2097152";
```
Comment (eonpatapon) [inline pkgs/docker-images/config/contrail/default.nix]: Use number

File: pkgs/docker-images/config/contrail/default.nix [hunk 10]
```diff
@@ -158,16 +159,25 @@ in rec {
...
           disc_server_ip = services.discovery.dns;
           disc_server_port = services.discovery.port;
 
+          vnc_connection_cache_size = 128;
+
           auth = "keystone";
           aaa_mode = "cloud-admin";
+          multi_tenancy             = "True";
+          list_optimization_enabled = "True";
+          apply_subnet_host_routes  = "True";
```
Comment (eonpatapon) [inline pkgs/docker-images/config/contrail/default.nix]: You can use booleans here I think

File: pkgs/docker-images/config/contrail/default.nix [hunk 11]
```diff
@@ -211,12 +239,17 @@ in rec {
...
+          sandesh_send_rate_limit = 100;
         }
         // logConfig services.svcMonitor
         // cassandraConfig
         // rabbitConfig
         // zookeeperConfig;
         KEYSTONE = keystoneConfig;
+        SCHEDULER = {
+          driver = "svc_monitor.scheduler.vrouter_scheduler.RandomScheduler";
+        };
```
Comment (eonpatapon) [inline pkgs/docker-images/config/contrail/default.nix]: Don't need to add this, this is the default

File: pkgs/docker-images/config/contrail/default.nix (hunk 8 above)
Comment (k-ayache) [reply pkgs/docker-images/config/contrail/default.nix]: In this case we can't use 86400 as number, it's very big number, it will be refused as number

Comment (nlewo) [reply pkgs/docker-images/config/contrail/default.nix]: what?

**Comment by nlewo** [comment]
Hydra is ok: http://hydra.int0.aub.cloudwatt.net/eval/733

**Pull request merged**